kratos/spatial_containers/search_wrapper.cpp

```diff
@@ -0,0 +1,473 @@
+    const double Radius,
+    ResultContainerType& rResults,
+    const bool SyncronizeResults
+    )
+{
+    // Search
+    std::vector<ResultType> results;
+    mpSearchObject->SearchInRadius(rPoint, Radius, results);
+    for (auto& r_result : results) {
+        rResults.AddResult(r_result);
+    }
+
+    // Synchronize if needed
+    if (SyncronizeResults) {
+        rResults.SynchronizeAll(mrDataCommunicator);
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TSearchObject, class TObjectType>
+void SearchWrapper<TSearchObject, TObjectType>::SerialSearchNearestInRadius(
+    const Point& rPoint,
+    const double Radius,
+    ResultContainerType& rResults,
+    const bool SyncronizeResults
+    )
+{
+    // Search
+    auto result = mpSearchObject->SearchNearestInRadius(rPoint, Radius);
+    rResults.AddResult(result);
+
+    // Synchronize if needed
+    if (SyncronizeResults) {
+        rResults.SynchronizeAll(mrDataCommunicator);
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TSearchObject, class TObjectType>
+void SearchWrapper<TSearchObject, TObjectType>::SerialSearchNearest(
+    const Point& rPoint,
+    ResultContainerType& rResults,
+    const bool SyncronizeResults
+    )
+{
+    // Search
+    auto result = mpSearchObject->SearchNearest(rPoint);
+    rResults.AddResult(result);
+
+    // Synchronize if needed
+    if (SyncronizeResults) {
+        rResults.SynchronizeAll(mrDataCommunicator);
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TSearchObject, class TObjectType>
+void SearchWrapper<TSearchObject, TObjectType>::SerialSearchIsInside(
+    const Point& rPoint,
+    ResultContainerType& rResults,
+    const bool SyncronizeResults
+    )
+{
+    // Search
+    auto result = mpSearchObject->SearchIsInside(rPoint);
+    rResults.AddResult(result);
+
+    // Synchronize if needed
+    if (SyncronizeResults) {
+        rResults.SynchronizeAll(mrDataCommunicator);
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+#ifdef KRATOS_USING_MPI
+
+template<class TSearchObject, class TObjectType>
+void SearchWrapper<TSearchObject, TObjectType>::DistributedSearchInRadius(
+    const Point& rPoint,
+    const double Radius,
+    ResultContainerType& rResults,
+    const bool SyncronizeResults
+    )
+{
+    // Check if the point is inside the set
+    if (SearchUtilities::PointIsInsideBoundingBox(mpSearchObject->GetBoundingBox(), rPoint, Radius)) {
```

Comment:
are u sure of this check? i would say that it is the bounding box increased by the radius all around